cmd/docker/builder.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strconv"
+
+	pluginmanager "github.com/docker/cli/cli-plugins/manager"
+	"github.com/docker/cli/cli/command"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const (
+	builderDefaultPlugin     = "buildx"
+	builderDefaultInstallMsg = `To install buildx, see https://docs.docker.com/go/buildx.`
+	builderErrorMsg          = `%s: Required builder component %s is missing or broken.`
+)
+
+type builderError struct {
+	warn    bool
+	builder string
+	err     error
+}
+
+func newBuilderError(warn bool, builder string, err error) error {
+	return &builderError{
+		warn:    warn,
+		builder: builder,
+		err:     err,
+	}
+}
+
+func (e *builderError) Error() string {
+	var errorMsg bytes.Buffer
+	if e.warn {
+		errorMsg.WriteString(fmt.Sprintf(builderErrorMsg, "WARNING", e.builder))
+	} else {
+		errorMsg.WriteString(fmt.Sprintf(builderErrorMsg, "ERROR", e.builder))
```

Comment:
Tempted to say; just concat strings here

```suggestion
		errorMsg.WriteString("WARNING: " + builderErrorMsg)
	} else {
		errorMsg.WriteString("ERROR: " + builderErrorMsg)
```

But perhaps we can simplify more; it looks overly complex 😅 (I'll push a branch with some ideas/thoughts)